src/openrct2/world/Map.cpp

```diff
@@ -4819,11 +4819,6 @@ void map_invalidate_virtual_floor_tiles()
         }
     }
 
-    // Do not invalidate if we're between ticks.
-    if (min_position.x == std::numeric_limits<sint16>::max()    || min_position.y == std::numeric_limits<sint16>::max() ||
-        max_position.x == std::numeric_limits<sint16>::lowest() || max_position.y == std::numeric_limits<sint16>::lowest())
```

Comment:
This was some very deliberate coding — removing it has a very real performance impact. Without it, FPS drops dramatically with the virtual floor active: down to 10fps (active) from 70fps (inactive) here.

Reply:
But does it actually solve the issue? I'll have a look at the performance issue, I haven't seen it on my end.

Reply:
Yes, removing it does solve the issue with rendering… If you could achieve that without the dip in FPS, I'll happily merge.